scripts/segmentation_HALO-20240912a.md

```diff
@@ -0,0 +1,410 @@
+# heading = ds["heading"].hvplot()
+# heading * takeoff * landing * \
+#  seg1s * seg1e * seg2s * seg2e * seg3s * seg3e * seg4s * seg4e * seg5s * seg5e * seg6s * seg6e * seg7s * seg7e * seg8s * seg8e * seg9s * seg9e * \
+#  seg10s * seg10e * seg11s * seg11e * seg12s * seg12e * seg13s * seg13e * seg14s * seg14e * seg15s * seg15e * seg16s * seg16e * seg17s * seg17e
+```
+
+```python jupyter={"outputs_hidden": true, "source_hidden": true}
+# roll = ds["roll"].hvplot()
+# roll * takeoff * landing * \
+#  seg1s * seg1e * seg2s * seg2e * seg3s * seg3e * seg4s * seg4e * seg5s * seg5e * seg6s * seg6e * seg7s * seg7e * seg8s * seg8e * seg9s * seg9e * \
+#  seg10s * seg10e * seg11s * seg11e * seg12s * seg12e * seg13s * seg13e * seg14s * seg14e * seg15s * seg15e * seg16s * seg16e * seg17s * seg17e
+```
+
+## Segments
+
+defined as a tuple of time slice (`start`, `end`) , segment `kind`, `name`, `remarks`.
+
+* in case of irregularities within a circle, 1 sec before the first and after the last sonde are chosen as start and end times
+* use the list of `remarks` to state any deviations, also with respective times
+
+Alternatively, you can also define the segments as dictionaries which also allows to add further attributes to single segments, e.g. a `radius` to a `circle` segment. At the end of the following code block all segments will be normalized by the `parse_segments` function.
+
+```python
+seg1 = (
+    slice("2024-09-12T11:33:02", "2024-09-12T11:46:57"),
+    ["straight_leg", "ascent"],
+    "eastward ascending ferry",
+)
+
+seg2 = (
+    slice("2024-09-12T11:52:44", "2024-09-12T12:01:28"),
+    ["straight_leg", "ascent"],
+    "eastward ascending ferry",
+
+)
+
+seg3 = (
+    slice("2024-09-12T12:01:28", "2024-09-12T12:35:09"),
+    ["straight_leg"],
+    "long eastward ferry",
+)
+
+seg4 = (
+    slice("2024--09-12T12:36:26", "2024-09-12T13:02:59"),
+    ["straight_leg"],
+    "leg to first circle"
+)
+
+seg5 = (
+    slice("2024-09-12T13:06:18", "2024-09-12T14:02:18"),
+    ["clockwise eastern circle"],
+    ["sonde failures"]
+)
+
+seg6 = (
+    slice("2024-09-12T14:06:42", "2024-09-12T14:12:06"),
+    ["straight_leg"],
+    "leg to second circle",
+)
+
+seg7 = (
+    slice("2024-09-12T14:18:20", "2024-09-12T15:10:37"),
+    ["circle"],
+    "counterclockwise eastern circle",
+)
+
+seg8 = (
+    slice("2024-09-12T15:13:08", "2024-09-12T15:18:19"),
+    ["straight_leg"],
+    "leg through circle",
+)
+
+seg9 = (
+    slice("2024-09-12T15:22:30", "2024-09-12T15:45:40"),
+    ["straight_leg"],
+    "leg to third circle",
+)
+
+seg10 = (
+    slice("2024-09-12T15:46:03", "2024-09-12T15:51:48"),
+    ["circle"],
+    "counterclockwise holding pattern",
+    ["small waiting circle"]
+)
+
+seg11 = (
+    slice("2024-09-12T15:56:35", "2024-09-12T17:03:46"),
+    ["circle"],
+    "counterclockwise western circle",
+)
+
+seg12 = (
+    slice("2024-09-12T17:12:26", "2024-09-12T17:53:12"),
+    ["straight_leg", "ec_track"],
```

Comment:
Did we decide to include `ec_underpass` for the segment on the ec_track that actually also includes the meeting point?
```suggestion
    ["straight_leg", "ec_track", "ec_underpass"],
```

Reply:
I would add it as a comment since ec_underpass is a kind of events and I think we should not confuse the categories..